src/main/java/com/farion/onlinebookstore/util/ConditionHolder.java

```diff
@@ -0,0 +1,14 @@
+package com.farion.onlinebookstore.util;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.stereotype.Component;
+
+@Setter
+@Getter
+@Component
+public class ConditionHolder {
+    private ThreadLocal<Set<Long>> condition = ThreadLocal.withInitial(HashSet::new);
```

Comment:
Also I did not get the point of having a common component for a whole application with a single thread-local var in it in the first place, because Spring has a separate thread to handle each request